libs/dh/wholesale/feature-calculations/src/lib/grid-areas/dropdown.component.ts

```diff
@@ -48,30 +47,36 @@ import { TranslocoDirective } from '@ngneat/transloco';
       [showResetOption]="false"
       [multiple]="true"
     >
-      <watt-field-hint>
-        {{ t('create.gridArea.hint', { count: control().value?.length }) }}
-      </watt-field-hint>
+      @if (isResolved()) {
+        <watt-field-hint>
+          {{ t('create.gridArea.hint', { count: control().value?.length }) }}
+        </watt-field-hint>
+      }
     </watt-dropdown>
   `,
 })
 export class DhCalculationsGridAreasDropdownComponent {
   featureFlags = inject(DhFeatureFlagsService);
 
   control = input.required<FormControl<string[] | null>>();
-  period = input.required<Range<string> | null>();
-  disabled = input(false);
+  period = input<WattRange<Date> | null>();
+
+  gridAreasQuery = lazyQuery(GetRelevantGridAreasDocument, { fetchPolicy: 'network-only' });
+  isResolved = computed(() => this.gridAreasQuery.status() === QueryStatus.Resolved);
 
-  gridAreasQuery = query(GetGridAreasDocument);
+  fetchGridAreas = effect(() => {
+    const period = this.period();
+    if (!period) this.gridAreasQuery.reset();
+    else this.gridAreasQuery.refetch({ period });
+  });
 
   gridAreas = computed(() => {
-    const gridAreas = this.gridAreasQuery.data()?.gridAreas ?? [];
+    const gridAreas = this.gridAreasQuery.data()?.relevantGridAreas ?? [];
 
     // HACK: This is a temporary solution to filter out grid areas that has no data
```

Comment:
Question: Is it still a hack when only relevantGridArea is fetched?

Reply:
Yes, that comment refers to the `gridAreas.filter((g) => ['803', '804', '533', '543', '584', '950']` line, which removes grid areas in the test environment that has no real data (and thus would cause calculations to fail). I haven't heard anything about this not being the case anymore, but I'll check. Would be nice to no longer need that hack 👍

Reply:
👍🏻

Reply:
Confirmed to still be required unfortunately